diff -ruN ufw-0.36.1.orig/conf/ufw.conf ufw-0.36.1/conf/ufw.conf
--- ufw-0.36.1.orig/conf/ufw.conf	2018-01-14 20:28:35.000000000 +0100
+++ ufw-0.36.1/conf/ufw.conf	2023-03-13 15:01:46.973221852 +0100
@@ -3,7 +3,7 @@
 
 # Set to yes to start on boot. If setting this remotely, be sure to add a rule
 # to allow your remote connection before starting ufw. Eg: 'ufw allow 22/tcp'
-ENABLED=no
+ENABLED=yes
 
 # Please use the 'ufw' command to set the loglevel. Eg: 'ufw logging medium'.
 # See 'man ufw' for details.
